src/util/source_location.h

```diff
@@ -188,4 +191,14 @@ class source_locationt:public irept
 
 std::ostream &operator <<(std::ostream &, const source_locationt &);
 
+template <>
+struct diagnostics_helpert<source_locationt>
+{
+  static std::string
+  diagnostics_as_string(const source_locationt &source_location)
+  {
+    return source_location.as_string();
```

Comment:
Maybe we can add a heading too, like `"source location:\n" + source_location.as_string()`.